packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+     * Note that the position is pre-scaling.
+     */
+    private getTransform(node: SKNode, desiredSize: number, canvasWidth: number, canvasHeight: number, scroll: Point, zoom: number): TransformAttributes {
+        // OLD: size dependant on node's bounds
+        // const proxyWidth = size * 0.001;
+        // const proxySizeScale = Math.min(proxyHeightScale, proxyWidthScale);
+        // console.log(proxySizeScale);
+
+        const pos = this.getPosition(node);
+        const bounds = node.bounds;
+
+        // Calculate the scale and the resulting proxy dimensions
+        // The scale is calculated such that width & height are capped to a max value
+        const proxyWidthScale = desiredSize / bounds.width;
+        const proxyHeightScale = desiredSize / bounds.height;
+        const scale = Math.min(1, proxyWidthScale, proxyHeightScale);
+        const proxyWidth = bounds.width * scale;
+        const proxyHeight = bounds.height * scale;
+
+        // Center at middle of node
+        const offsetX = 0.5 * (node.bounds.width * zoom - proxyWidth);
+        const offsetY = 0.5 * (node.bounds.height * zoom - proxyHeight);
+        let x = (pos.x - scroll.x) * zoom + offsetX;
+        let y = (pos.y - scroll.y) * zoom + offsetY;
+
+        // Cap proxy at canvas border
+        x = Math.max(0, Math.min(canvasWidth - proxyWidth, x));
+        y = Math.max(0, Math.min(canvasHeight - proxyHeight, y));
+
+        // Make sure the proxies aren't rendered behind the sidebar buttons at the top right
+        /* Don't need to check for the opened sidebar since it closes as soon as the diagram is moved
+          (onMouseDown), e.g. don't reposition proxies accordingly */
+        const rect = document.querySelector('.sidebar__toggle-container')?.getBoundingClientRect();
+        const isSidebarOpen = document.querySelector('.sidebar--open');
+        if (!isSidebarOpen && rect && y < rect.bottom && x > rect.left - proxyWidth) {
+            x = rect.left - proxyWidth;
+        }
+
+        if (this.capProxyToParent && node.parent && node.parent.id !== "$root") {
+            const parent = node.parent as SKNode;
+            const parentBounds = parent.bounds;
+            const parentPos = this.getPosition(parent);
+            x = Math.max((parentPos.x - scroll.x) * zoom, Math.min((parentPos.x + parentBounds.width - scroll.x) * zoom - proxyWidth, x));
+            y = Math.max((parentPos.y - scroll.y) * zoom, Math.min((parentPos.y + parentBounds.height - scroll.y) * zoom - proxyHeight, y));
+        }
+
+        // OLD: Scale the coordinates (to get position post-scaling)
+        // x /= scale;
+        // y /= scale;
+
+        return { x: x, y: y, scale: scale, width: proxyWidth, height: proxyHeight };
+    }
+
+    /** Recursively calculates the positions of this node and all of its predecessors and stores them in {@link positions}. */
+    private getPosition(node: SKNode): Point {
+        if (!node) {
+            return { x: 0, y: 0 };
```

Comment:
if something undefined is passed here, wouldn't it make more sense to return undefined? this could promote misusing the function